src/GrpcProtos/capture.proto

```diff
@@ -521,6 +521,7 @@ message ProducerCaptureEvent {
     InternedCallstack interned_callstack = 7;
     InternedString interned_string = 18;
     IntrospectionScope introspection_scope = 19;
+    MemoryEventWrapper memory_event_wrapper = 29;
```

Comment:
This is just a question: Why are the "old" memory events removed from `ClientCaptureEvent`, but not from ProducerCaptureEvent`?

Reply:
This makes me realize that I'm not very familiar with this code and probably can't review it properly. I'm not sure how familier @karupayun is with this. So maybe we should add another reviewer? Davide for example? Pablo, WDYT?

Reply:
Removing old memory events from `ProducerCpatureEvent` and updating the integration test need to be done at the same commit. So I put this part in a later commit which updates the integration test.

Reply:
I'm seeing that Davide is also a reviewer now.